From dffff93445721d231c87e92f61d8afe214b653a7 Mon Sep 17 00:00:00 2001
From: Stefan Szabo <szbs001@freescale.com>
Date: Thu, 17 May 2012 14:21:49 +0000
Subject: [PATCH 10/19] powerpc/corenet: 14 Gb/s configuration

should be squashed into commit: 83a4df4a4b306569d1b17575adc94d8138235719
powerpc/corenet: 14 Gb/s configuration

New default FMan configuration (14G/FMan):
* Enable out-of-the-box the automatic FMan resource (FIFOs, TNUMs, DMAs)
  allocation configuration,
* disable one 1Gb Ethernet port of each FMan on P3/5 platforms
  (one that will -- in FSL-shipped configurations -- likely be NC,
   or connected to an SGMII),
* disable three of the Offline Parsing ports (save for O/H port #0,
  which is required for host commands),
This is the maximal configuration that supports jumbo frames out of the box
(with jumbo NOT enabled out-of-the-box).

Signed-off-by: Stefan Szabo <szbs001@freescale.com>
[Yongli: Original patch taken from FSL
    QorIQ-SDK-V1.2-SOURCE-20120614-yocto.iso image]
Integrated-by: Yongli He <yongli.he@windriver.com>
---
 arch/powerpc/boot/dts/p5020ds.dts |    3 +++
 arch/powerpc/boot/dts/p5040ds.dts |    2 ++
 2 files changed, 5 insertions(+)

diff --git a/arch/powerpc/boot/dts/p5020ds.dts b/arch/powerpc/boot/dts/p5020ds.dts
index c13f4a1..f914600 100644
--- a/arch/powerpc/boot/dts/p5020ds.dts
+++ b/arch/powerpc/boot/dts/p5020ds.dts
@@ -151,6 +151,7 @@
 					#size-cells = <0>;
 					compatible = "fsl,hydra-mdio";
 					fsl,mdio-handle = <&mdio0>;
+					fsl,hydra-mdio-mux-mask = <0x70>;
 					fsl,hydra-mdio-muxval = <0x00>;
 					status = "disabled";
 
@@ -176,6 +177,7 @@
 					#size-cells = <0>;
 					compatible = "fsl,hydra-mdio";
 					fsl,mdio-handle = <&mdio0>;
+					fsl,hydra-mdio-mux-mask = <0x70>;
 					fsl,hydra-mdio-muxval = <0x00>;
 					status = "disabled";
 
@@ -427,6 +429,7 @@
 		ethernet@2 {
 			compatible = "fsl,p5020-dpa-ethernet", "fsl,dpa-ethernet";
 			fsl,fman-mac = <&enet2>;
+			status = "disabled";
 		};
 		ethernet@3 {
 			compatible = "fsl,p5020-dpa-ethernet", "fsl,dpa-ethernet";
diff --git a/arch/powerpc/boot/dts/p5040ds.dts b/arch/powerpc/boot/dts/p5040ds.dts
index 9bee35b..58f02c6 100644
--- a/arch/powerpc/boot/dts/p5040ds.dts
+++ b/arch/powerpc/boot/dts/p5040ds.dts
@@ -561,6 +561,7 @@
 		ethernet@0 {
 			compatible = "fsl,p5040-dpa-ethernet", "fsl,dpa-ethernet";
 			fsl,fman-mac = <&fm1dtsec1>;
+			status = "disabled";
 		};
 		ethernet@1 {
 			compatible = "fsl,p5040-dpa-ethernet", "fsl,dpa-ethernet";
@@ -585,6 +586,7 @@
 		ethernet@6 {
 			compatible = "fsl,p5040-dpa-ethernet", "fsl,dpa-ethernet";
 			fsl,fman-mac = <&fm2dtsec1>;
+			status = "disabled";
 		};
 		ethernet@7 {
 			compatible = "fsl,p5040-dpa-ethernet", "fsl,dpa-ethernet";
-- 
1.7.9.7

